### PR TITLE
Feature/form

### PR DESCRIPTION
**Pull request by ramirobg94**

### CODE REVIEW

File: src/screens/ConnectionSettings/screen.js [hunk 1]
```diff
@@ -69,14 +69,18 @@ const ConnectionSettings = () => {
...
+        const qrDataParsed = JSON.parse(qrData);
+
+        setFormData(prevData => ({
+          ...prevData,
+          username: qrDataParsed?.username || prevData.username,
+          serverUrl: qrDataParsed?.serverUrl || prevData.serverUrl,
+          password: qrDataParsed?.password || prevData.password,
+        }));
+      } catch (e) {
+        console.log('qrData is not a valid json');
```
Comment (SteRiccio) [inline src/screens/ConnectionSettings/screen.js:82]: is there a better logging strategy we can use or put in place?

File: src/state/sagas.js [hunk 2]
```diff
@@ -12,4 +13,5 @@ export default function* rootSaga() {
   yield all([fork(survey)]);
   yield all([fork(records)]);
   yield all([fork(nodes)]);
+  yield all([fork(form)]);
```
Comment (SteRiccio) [inline src/state/sagas.js:16]: is it necessary to use yield all even if the array param has only one element?

File: src/state/surveys/api/api.js [hunk 3]
```diff
@@ -8,7 +8,9 @@ const getSurveys = async ({serverUrl}) => {
 
 const getSurveyById = async ({serverUrl, surveyId}) => {
   const {data: survey} = await API({serverUrl}).get({
-    path: `api/survey/${surveyId}`,
+    path: `api/survey/${surveyId}?${new URLSearchParams({
```
Comment (SteRiccio) [inline src/state/surveys/api/api.js:11]: maybe the parameters could be passed as an object to the API.get method, so the use of URLSearchParams could be moved there...

**Review by SteRiccio — APPROVED** [review]

File: src/screens/ConnectionSettings/screen.js (hunk 1 above)
Comment (ramirobg94) [reply src/screens/ConnectionSettings/screen.js:82]: Idk maybe console.error

File: src/state/sagas.js (hunk 2 above)
Comment (ramirobg94) [reply src/state/sagas.js:16]: I used to initialize in this way maybe we can put togthr

File: src/state/surveys/api/api.js (hunk 3 above)
Comment (ramirobg94) [reply src/state/surveys/api/api.js:11]: Yep I have to update the internal method to get params

File: src/screens/ConnectionSettings/screen.js (hunk 1 above)
Comment (SteRiccio) [reply src/screens/ConnectionSettings/screen.js:82]: maybe the user should be informed in this case, when something went wrong, with a toast or something like this

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**